### DISCUSSION TITLE
Is shared memory transports not enabled on Android?

### DISCUSSION THREAD

**Discussion by niceasia** [OpenDDS/OpenDDS#4204] [Q&A]
Hello, I want to use shared memory transports on the Android platform, but the error message shows that the function is not enabled. I checked the latest version of the development document and it is also mentioned that it is not enabled. Is there a corresponding enabling method or is there a plan to enable shared memory transports? thanks！
![15343ca1-b740-4def-8d12-2d853a287ebb](https://github.com/OpenDDS/OpenDDS/assets/115396816/efbb270e-227e-4cd5-a9b3-e1bafe365ba2)

**Comment by mitza-oci** [comment]
Could you put a breakpoint on that ACE_Mutex constructor and post a stack trace so we can see where it's being used?

**Reply by niceasia** [reply]
I searched all the codes, but did not find the location where the log reported the error, so is shared memory transports feasible on the Android platform? Thanks！
![image](https://github.com/OpenDDS/OpenDDS/assets/115396816/f405eacb-c522-4945-acc1-e673fcb36498)

**Reply by mitza-oci** [reply]
ACE_Mutex is a class in the ACE library github.com/DOCGroup/ACE_TAO
